lib/interactivity-api.php

```diff
@@ -16,7 +16,7 @@ function gutenberg_reregister_interactivity_script_modules() {
 
 	wp_register_script_module(
 		'@wordpress/interactivity',
-		gutenberg_url( '/build/interactivity/index.min.js' ),
+		gutenberg_url( '/build/interactivity/' . ( wp_is_development_mode( 'plugin' ) ? 'debug.min.js' : 'index.min.js' ) ),
```

Comment:
Probably we want to include here other development modes (`theme`, `core`). Or simply run:
```php
isset( WP_DEVELOPMENT_MODE ) ? 'debug.min.js' : 'index.min.js'
```

There's also `WP_DEBUG` to indicate that WordPress should run in debug mode. Would it make sense to use that constant instead? 🤔 

cc: @gziolo

Reply:
It seems like any development mode could potentially benefit from this. Certainly `core` and `plugin` at this time.

Reply:
`wp_is_development_mode` is used mostly to disable caching in PHP. More details in https://make.wordpress.org/core/2023/07/14/configuring-development-mode-in-6-3/.

Here is the list of constants you can use with assets:

https://github.com/WordPress/wordpress-develop/blob/18db904c293f43f5e928b74835d0969cd60c1e2e/src/wp-includes/script-loader.php#L5-L10

```
define('SCRIPT_DEBUG', true); loads the development (non-minified) versions of all scripts and CSS, and disables compression and concatenation,
```

By the way, Interactivity API scripts in the Gutenberg plugin are incorrectly always registered using the production version at the moment. It was fixed in WP core:

https://github.com/WordPress/wordpress-develop/blob/18db904c293f43f5e928b74835d0969cd60c1e2e/src/wp-includes/interactivity-api/class-wp-interactivity-api.php#L188-L201

The version in WP core uses the helper function `wp_scripts_get_suffix` that sets the `.js` or `.min.js` using a few conditions where check for `SCRIPT_DEBUG` is one of them.

The same constant is being injected to JS using webpack, so you can also used it to enforce dead code elimination on production when using debug tools:

https://github.com/WordPress/gutenberg/blob/94eaca9d6cf872980e6517ce97a6b2d87b0f7c3b/tools/webpack/shared.js#L72-L73

Reply:
OK, let's use `SCRIPT_DEBUG` then.

Reply:
@gziolo, regarding what you mentioned about Gutenberg always registering the production version of the Interactivity API scripts, is that something we can fix in this PR?

Reply:
I think we can add the suffix variable, similar to what is done in Core.